lib/OpenQA/Setup.pm

```diff
@@ -220,6 +224,38 @@ sub update_config {
     }
 }
 
+sub prepare_settings_ui_keys {
+    my ($app) = shift;
+    my $keys = $app->config->{job_settings_ui}->{keys_to_render_as_links};
+    my $settings_ui_cfg;
+    my @link_settings;
+    my @link_keys    = split(/,/, $keys);
+    my $cfgpath      = $ENV{OPENQA_CONFIG} ? path($ENV{OPENQA_CONFIG}) : $app->home->child("etc", "openqa");
+    my $linkkeysfile = $cfgpath->child('settings_ui_links.ini');
+
+    if (-e $linkkeysfile) {
+        $settings_ui_cfg = Config::IniFiles->new(-file => $linkkeysfile->to_string) || undef;
+        $app->config->{link_config} = $settings_ui_cfg;
+
+    }
+    else {
+        $app->log->warn("No configuration file supplied for settings_ui links in $cfgpath");
+        return;
+    }
```

Comment:
Please just make the case of the file non-existing and early-return, e.g. `$app->log->warn("…") and return unless -e $linkkeysfile`.

Another recommendation: Use an underscore-separated variable name, e.g. `link_keys_file`

Reply:
done